### ISSUE TITLE
Multiple small/nonstandard instances not supported

### ISSUE THREAD

**Issue by andrewstuart** [laurencedawson/sync-for-lemmy#418]
**Describe the bug**  
I have a couple small instances I use and manage, but with Sync (and also with Jerboa) I can only log into one unlisted Lemmy instance at a time. If I try to add `b`, `a` disappears. The settings seem to still stick around, but the instance is gone from the switcher, replaced by `b`.

**Steps to Reproduce**  
1. Go to Add Account
2. Add custom instance not in the drop-down
3. Use the instance to validate functionality. Works fine.
4. Go to Add Account
5. Add another custom instance not in the drop-down.
6. Validate this instance works too.
7. Open the account picker again, and observe that step 2 is gone, replaced by step 5.
8. [Optional] Re-add original instance and skip setup. Config will be as originally set up.

**Expected behavior**  
Both instances should be available in the drop-down for fast switching, as currently seems to work with listed instances plus up to one custom instance.

**Smartphone (please complete the following information):**

**Device information**

    Sync version: v23.08.25-12:16    
    Sync flavor: googlePlay    
    
    Ultra user: true    
    View type: List    
    Push enabled: false    
    
    Device: raven    
    Model: Google Pixel 6 Pro    
    Android: 13

**Comment by laurencedawson** [comment]
Very odd, the size of the instance shouldn't matter at all here.

Do they have public signups on these instances so I can try to replicate this behaviour?

Thanks

**Comment by andrewstuart** [comment]
Yeah they're open, just with captcha+approval, but I can approve you quickly if you sign up.

lemmy.stuart.fun and lululemmy.com are the two I've replicated this with, both hosted by me.

**Comment by andrewstuart** [comment]
Okay, I think I figured out the issue. Looks like my `sub` comes back in the JWT as the same integer for both instances, and that causes some kind of havoc. Jerboa has this issue too, and behaves pretty oddly.  I created another username for myself on one of my instances and it just works as expected now.
